### PR TITLE
Port to preCICE v3 API

### PR DESCRIPTION
**Pull request by MakisH**
Updates the Fortran module to preCICE v3 (https://github.com/precice/precice/pull/1914).

- [x] Port interface to v3 API
- [x] Port solverdummy code to v3 interface
- [x] Port solverdummy config to v3 interface
- [x] Debug

@fsimonis Please check in particular whether I have missed defining anything as an array (pointer).

I think there were already many ghosts in the interface file, and the fact that the interface is not checked for linking to preCICE does not make me feel very comfortable. Maybe there is a way to check that.

### CODE REVIEW

File: precice.f03 [hunk 1]
```diff
@@ -38,458 +31,284 @@ subroutine precicef_advance(timestepLengthLimit) &
...
 
-    subroutine precicef_set_vertex(meshID, position, vertexID) &
+    subroutine precicef_set_vertex(meshName, position, vertexID) &
       &  bind(c, name='precicef_set_vertex_')
 
       use, intrinsic :: iso_c_binding
-      integer(kind=c_int) :: meshID
-      real(kind=c_double) :: position(3)
-      integer(kind=c_int) :: vertexID
+      character(kind=c_char), dimension(*) :: meshName
```
Comment (fsimonis) [inline precice.f03:110]: Length is missing

Comment (MakisH) [reply precice.f03:110]: Good catch, fixed, thanks!

I guess we need kind of an "extended solver dummy" that only calls all functions to make sure that they are all correct and link to preCICE.

Or somehow check the linking during compilation of the module.

Comment (uekerman) [reply precice.f03:110]: The mocked interface (workpackage of preECO) should ultimately be a solution for this problem. Then, such a extended solverdummy can run alone and also test the API, i.e. it becomes a normal test "main".

Comment (MakisH) [reply precice.f03:110]: I had a simpler usecase in mind (just that the interface is correct), which would probably be solvable in a simpler way, but having a mocked interface would also of course help.

Note that the problem here is not that we cannot call preCICE. It is that the building procedure of the interface does not try to link to preCICE.

Comment (uekerman) [reply precice.f03:110]: Yes, I get that. With the mocked interface, we would then test the linking when running the tests. I am not sure if there is a simpler solution?

Comment (uekerman) [reply precice.f03:110]: Problem with a solverdummy is that you need to use all API functions in a meaningful "integration-test-like" setting. This gets complicated ...
With the mocked interface, we could test each API function in an isolated setup.

Comment (MakisH) [reply precice.f03:110]: But not if we don't ever run the built executable, right?

Comment (fsimonis) [reply precice.f03:110]: If the linking fails when the mod is incorrect, then a mocked interface makes sense. Is that the case though?

Comment (MakisH) [reply precice.f03:110]: Let's continue the discussion in this issue: https://github.com/precice/fortran-module/issues/15

**Pull request merged**